query-engine/connector-test-kit-rs/query-engine-tests/tests/new/limits.rs

```diff
@@ -0,0 +1,90 @@
+use query_engine_tests::*;
+
+// NOTE: This test depends on the absence of the env variable QUERY_BATCH_SIZE
```

Comment:
Then this test is not useful unfortunately. Also, I don't really understand what you're trying to test that is not covered by the current query splitting test. I understand that you're verifying that this prevents us from going over the native mysql limit, but that will never be run in CI or usual dev setups. All it does is taking ages without adding real value.

Reply:
Heads-up here: I suggested adding integration tests as there clearly was something not caught up by the existing tests since MySQL was failing on a certain amount of param biding (hence this PR)